File: Rentalaja-be/app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    protected $fillable = [
        'user_id', 'product_id', 
        'start_date', 'end_date', 'pickup_time', 'return_time', 
        'total_price', 'status', 'payment_method', 'payment_proof'
    ];

    protected function paymentProof(): \Illuminate\Database\Eloquent\Casts\Attribute
    {
        return \Illuminate\Database\Eloquent\Casts\Attribute::make(
            get: fn ($value) => $value ? url('storage/' . $value) : null,
        );
    }
    
    public function product() { 
        return $this->belongsTo(Product::class); 
    }

    public function user() {
        return $this->belongsTo(User::class);
    }
}

File: Rentalaja-fe/lib/features/catalogue/domain/transaction.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'electronic_item.dart';

part 'transaction.freezed.dart';
part 'transaction.g.dart';

@freezed
class Transaction with _$Transaction {
  const factory Transaction({
    required int id,
    @JsonKey(name: 'user_id') required int userId,
    @JsonKey(name: 'product_id') required int productId,
    @JsonKey(name: 'start_date') required DateTime startDate,
    @JsonKey(name: 'end_date') required DateTime endDate,
    @JsonKey(name: 'pickup_time') required String pickupTime,
    @JsonKey(name: 'return_time') required String returnTime,
    @JsonKey(name: 'total_price') required double totalPrice,
    required String status,
    @JsonKey(name: 'payment_method') required String paymentMethod,
    @JsonKey(name: 'payment_proof') String? paymentProof,
    // Product relationship (loaded via with('product'))
    ElectronicItem? product,
    @JsonKey(name: 'is_rated') bool? isRated,
  }) = _Transaction;

  factory Transaction.fromJson(Map<String, dynamic> json) =>
      _$TransactionFromJson(json);
}
